syntax = "proto3";

service PortMapper {
    rpc RegisterService (RegisterRequest) returns (RegisterReply) {}

    rpc UnregisterService (UnregisterRequest) returns (UnregisterReply) {}

    rpc GetServiceInfo (GetInfoRequest) returns (GetInfoReply) {}

    rpc Terminate (TerminateRequest) returns (TerminateReply) {}
}

message RegisterRequest {
    string name = 1;
    float value = 2;
}

message RegisterReply {
    int32 port = 1;
}

message UnregisterRequest {
    int32 port = 1;
}

message UnregisterReply {
    int32 status = 1;
}

message GetInfoRequest {
    string name = 1;
}

message GetInfoReply {
    int32 port = 1;
    float value = 2;
}

message TerminateRequest {

}

message TerminateReply {
    int32 registered_services_count = 1;
}
